src/testshade/batched_simplerend.cpp

```diff
@@ -391,31 +391,32 @@ BatchedSimpleRenderer<WidthT>::trace(
 template<int WidthT>
 void
 BatchedSimpleRenderer<WidthT>::getmessage(BatchedShaderGlobals* bsg,
-                                          Masked<int> result, ustring source,
-                                          ustring name, MaskedData val)
+                                          Masked<int> result,
+                                          ustringhash source, ustringhash name,
+                                          MaskedData val)
 {
-    OSL_ASSERT(source == ustring("trace"));
+    OSL_ASSERT(source == "trace");
```

Comment:
Do we know when "trace" is converted to a hash code?  It looks like from ustringhash::operator ==.  Can that be made constexpr?

Reply:
I think it can, on the oiio side.

Reply:
OK, think ustringhash::ustringhash constructor can/should be constexpr as well.

Reply:
it can't be, because it constructs a ustring. ustringhash is ONLY supposed to hold the hash of an actual ustring, not just any size_t.